lib/ProductOpener/Data.pm

```diff
@@ -119,6 +122,43 @@ sub execute_count_tags_query ($query) {
 	return execute_tags_query('count', $query);
 }
 
+sub execute_product_query ($parameters_ref, $query, $fields, $sort, $limit, $skip) {
+	# Convert sort into an array so that the order of keys is not ambiguous
+	my @sort_array = ();
+	if ($sort) {
+		foreach my $k ($sort->Keys) {
+			push(@sort_array, [$k, $sort->FETCH($k)]);
+		}
+	}
+
+	my $results = execute_tags_query(
```

Comment:
Shouldn't we add a check that the sort is by popularity, so that we don't send queries to off-query if we already know they won't be handled?

Or maybe we could add sorting options for created_t, last_modified_t, Nutri-Score and Green-Score :)

Reply:
I was thinking we would send all queries so that off-query can be enhanced to support more without needing to make further changes to Product Opener. If off-query can't handle the query it forwards it to MongoDB anyway so there is an extra network hop but hopefully shouldn't have a big impact on performance

Reply:
I changed tack and have now restricted it to only send queries that are sorted Hy popilarity_key for now